spec/models/ems_event_spec.rb

```diff
@@ -196,6 +198,24 @@
           expect(new_event.availability_zone_id).to eq @availability_zone.id
         end
       end
+
+      context "when two events have the same timestamp" do
+        context "when events belong to the same object" do
+          it "should add only one of them" do
+            EmsEvent.add(@ems.id, @event_hash)
+            second = EmsEvent.add(@ems.id, @event_hash)
+            expect(second).to be_nil
```

Comment:
Should we check that there is only one EmsEvent row for the event hash details instead of checking the return value from `add`?